test/jdk/java/security/Provider/ChangeProviders.java

```diff
@@ -81,6 +81,22 @@ public static void main0(String[] args) throws Exception {
             throw new Exception("Provider not at pos 1");
         }
 
+        //Ensure that providers inserted at position 0 or > n are placed
```

Comment:
nit: this sentence is a bit hard to parse, maybe it's easier to just state that positions out of the [1...n] range is placed at the position n+1?

Reply:
Actually, you don't need another provider impl, you could have just removed Foo and then proceed with the Foo provider with your insertProviderAt(...) testing. With the current test, when you start testing, the provider count is n+1, which can be confusing mixed with your comment of using n as the current provider count... Thus, I'd suggest you just remove the Foo provider and re-use it. This way, there is less code and all is consistent.